analyzer_test.go

```diff
@@ -48,6 +48,66 @@ func header(header string) *goheader.Target {
 		Path: os.TempDir(),
 	}
 }
+
+func TestAnalyzer_Analyze6(t *testing.T) {
+	a := goheader.New(
+		goheader.WithTemplate("A {{ some-value }} B"),
+		goheader.WithValues(map[string]goheader.Value{
+			"SOME-VALUE": &goheader.ConstValue{
+				RawValue: "{{ some-value }}",
+			},
+		}),
+	)
+	var issue = a.Analyze(header("A {{ SOME-VALUE }} B"))
+	require.NotNil(t, issue)
+	require.Contains(t, issue.Message(), "recursion detected")
+	issue = a.Analyze(header("A {{ SOME-VALUE }} C"))
```

Comment:
cosmetic: something I tried to fix in the last MR - the lack of whitespaces makes the tests difficult to understand